### ISSUE TITLE
map from ClinVar UID to a MeSH UID

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#31]
Originally proposed by @dkoslicki:

> Related question: is there any way to convert between clinvarid and meshid/term? I see get_clinvar_uids_for_disease_or_phenotype_string, but I don't see any method to convert clinvar to mesh.

Steve says:

> It appears not, but it looks like we can connect from ClniVar OrphaNet and from OrphaNet we can get a MeSH ID.

**Comment by finnagin** [comment]
I added a method that does this to QueryNCBIeUtils.py but it needs more testing and I think it might be a good idea to add more pathways for conversion. right now it tries: ClinVar -> MeSH then ClinVar -> MedGen -> MeSH. Adding more might make it more robust.

However, This might not be needed currently since I also added a method that converts directly from phenotype/disease name into MeSH ids.

**Comment by dkoslicki** [comment]
@finnagin What's the status of this issue? In light of #38, there seems to be multiple MeSH stuff that still needs to be sorted. Is that correct?

**Comment by finnagin** [comment]
@dkoslicki My understanding was that the reason we needed a conversion from Cinvar -> MeSH was that we trying to go from <phenotype/disease string> -> Clinvar -> MeSH but I found a way to go from <phenotype/disease string> -> Mesh without having to get the Clinvar id first. (which I added as the method `get_mesh_uids_for_disease_or_phenotype_string` in QueryNCBIeUtils.py) I thought that this solved the issue that we were using the Clinvar ids for.

However, I had also added a method that converts from Clinvar -> MeSH by first trying a direct conversion then trying to convert to Cilnvar -> medgen -> MeSH. If we are going to use this method I mentioned that I think that it would be useful to add more indirect conversions if the current two fail but I haven't worked on it since my understanding was that `get_mesh_uids_for_disease_or_phenotype_string` circumvents the need for a method to convert from Clinvar -> MeSH. If I am wrong about this then I can get to work on trying to make this method more robust.
